utils/transaction-provider/transaction-provider.ts

```diff
@@ -0,0 +1,64 @@
+import { type WalletCore } from "@trustwallet/wallet-core";
+import type { CoinType } from "@trustwallet/wallet-core/dist/src/wallet-core";
+
+import { ChainKey, Currency, rpcUrl } from "~utils/constants";
+import ThorchainTransactionProvider from "./thorchain/thorchain-tx-provider";
+import EVMTransactionProvider from "./evm/evm-tx-provider";
+import MayaTransactionProvider from "./maya/maya-tx-provider";
+import GaiaTransactionProvider from "./gaia/gaia-tx-provider";
+
+interface ChainRef {
+  [chainKey: string]: CoinType;
+}
+
+export default class TransactionProvider {
+  static createProvider(
+    chainKey: ChainKey,
+    chainRef: ChainRef,
+    dataEncoder: (data: Uint8Array) => Promise<string>,
+    walletCore: WalletCore
+  ) {
+    switch (chainKey) {
+      case ChainKey.THORCHAIN: {
+        return new ThorchainTransactionProvider(
+          chainKey,
+          chainRef,
+          dataEncoder,
+          walletCore
+        );
+      }
+      case ChainKey.MAYACHAIN: {
+        return new MayaTransactionProvider(
+          chainKey,
+          chainRef,
+          dataEncoder,
+          walletCore
+        );
+      }
+      case ChainKey.GAIACHAIN: {
+        return new GaiaTransactionProvider(
+          chainKey,
+          chainRef,
+          dataEncoder,
+          walletCore
+        );
+      }
+      case ChainKey.OPTIMISM: {
```

Comment:
why `ChainKey.OPTIMISM` is using `GaiaTransactionProvider` ? it should be EVMTransactionProvider